### DISCUSSION TITLE
Share, how to get it?

### DISCUSSION THREAD

**Discussion by x1arch** [opensubsonic/open-subsonic-api#46] [General]
We have a four methods for sharing music:

- [getShares](https://opensubsonic.netlify.app/docs/endpoints/getshares) 
- [createShare](https://opensubsonic.netlify.app/docs/endpoints/createshare)
- [updateShare](https://opensubsonic.netlify.app/docs/endpoints/updateshare)
- [deleteShare](https://opensubsonic.netlify.app/docs/endpoints/deleteshare)

BUT! We have no method for getting this share, how it's possible? I check many servers and only Navidome implement `shares`, but they get data for share by there internal methods. Why? Maybe I do not understand something?

At responses section I see response for [share](https://opensubsonic.netlify.app/docs/responses/share/) but I do not see the method which will return this response. PS I found and `share` it's part of `shares` result, sadly.

**Comment by x1arch** [comment]
Why we can't add new items to share? Update method allow update only description and expiration but do not items, why?
Why we have no security option for share (when create or update)? If I wants share it only for peoples who knows the password (secret phrase).

Because only Navidome implement this methods we can fix it do not worry break something.

**Reply by lachlan-00** [reply]
Create a new share

**Reply by x1arch** [reply]
Yes, it was my first idea, but it not helps, because `id` of share will be change. If this share for many peoples, all should will get the new url, it can be problem. Add two optional params to request, as `add` and `remove`, are more easy and do not break compatibility.

PS I suppose we should add one more param which will allow or disallow download tracks from share.

**Reply by lachlan-00** [reply]
i would just add to updateshare and allow a new list of items in the share if you wanted to keep the id

but i also don't really think shares should be thought of as a permanent object where you'd care about the id.

**Reply by epoupon** [reply]
This feels weird to update a share, as you likely still need to tell your users you updated it (in that case, this is close to sending a brand new share)

**Reply by x1arch** [reply]
Absolutely not, every reopen browser window (or page refresh) will update playlist and i do not must notify anybody about that. Imagine, I will share playlist with some background music for a few points (hostels, restaurants, etc.) and then i want add another compositions to this playlist, and i should share new link for every point, it's not a problem when it 1-2 points, but when we are talking about 100 points? Share as a podcast, you do not need new link for get every new episode (as example I can share podcast episodes and add new episodes, when it will available).

PS. When you use share in google drive or dropbox, update share is do not weird?!
